.cicd/scripts/srw_ftest.sh

```diff
@@ -0,0 +1,137 @@
+[[ -n ${BRANCH} ]] || BRANCH="develop"
+[[ -n ${TASKS} ]] || TASKS=""
+[[ -n ${TASK_DEPTH} ]] || TASK_DEPTH=4
+[[ -n ${FORGIVE_CONDA} ]] || FORGIVE_CONDA=true
+set -e -u -x
+
+script_dir="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" > /dev/null 2>&1 && pwd)"
+
+# Get repository root from Jenkins WORKSPACE variable if set, otherwise, set
+# relative to script directory.
+declare workspace
+if [[ -n "${WORKSPACE}" ]]; then
+    workspace="${WORKSPACE}"
+else
+    workspace="$(cd -- "${script_dir}/../.." && pwd)"
+fi
+
+# Normalize Parallel Works cluster platform value.
+declare platform
+if [[ "${SRW_PLATFORM}" =~ ^(az|g|p)clusternoaa ]]; then
+    platform='noaacloud'
+else
+    platform="${SRW_PLATFORM}"
+fi
+
+# Test directories
+we2e_experiment_base_dir="${workspace}/expt_dirs"
+we2e_test_dir="${workspace}/tests/WE2E"
+nco_dir="${workspace}/nco_dirs"
+
+pwd
+
+# The following essentially sets up and performs tasks described in the user documentation section:
+#     "Run the Workflow Using Stand-Alone Scripts".
+echo "BRANCH=${BRANCH}"
+
+# Set the ACCOUNT to use for this PLATFORM ...
+sed "s|^  ACCOUNT: \"\"|  ACCOUNT: \"${ACCOUNT}\"|1" -i ush/config_defaults.yaml
+sed "s|hera|${platform,,}|1" ush/config.community.yaml | sed "s|an_account|${ACCOUNT}|1" > ush/config.yaml
+
+# Set directory paths ...
+export EXPTDIR=${workspace}/expt_dirs/test_community
+echo "EXPTDIR=${EXPTDIR}"
+sed "s|^workflow:|workflow:\n  EXPT_BASEDIR: ${workspace}/expt_dirs|1" -i ush/config.yaml
+sed "s|^workflow:|workflow:\n  EXEC_SUBDIR: ${workspace}/install_${SRW_COMPILER}/exec|1" -i ush/config.yaml
+
+# DATA_LOCATION differs on each platform ... find it.
+export DATA_LOCATION=$(grep TEST_EXTRN_MDL_SOURCE_BASEDIR ${workspace}/ush/machine/${SRW_PLATFORM,,}.yaml | awk '{printf "%s", $2}')
+echo "DATA_LOCATION=${DATA_LOCATION}"
+
+# Configure a default test ...
+sed "s|^task_get_extrn_ics:|task_get_extrn_ics:\n  EXTRN_MDL_SOURCE_BASEDIR_ICS: ${DATA_LOCATION}/FV3GFS/grib2/2019061518|1" -i ush/config.yaml
+sed "s|^task_get_extrn_lbcs:|task_get_extrn_lbcs:\n  EXTRN_MDL_SOURCE_BASEDIR_LBCS: ${DATA_LOCATION}/FV3GFS/grib2/2019061518|1" -i ush/config.yaml
+
+# Activate the workflow environment ...
+source etc/lmod-setup.sh ${platform,,}
+module use modulefiles
+module load build_${platform,,}_${SRW_COMPILER}
+module load wflow_${platform,,}
+
+[[ ${FORGIVE_CONDA} == true ]] && set +e +u    # Some platforms have incomplete python3 or conda support, but wouldn't necessarily block workflow tests
+conda activate regional_workflow
+set -e -u
+
+cd ${workspace}/ush
+        # Consistency check ...
+        ./config_utils.py -c ./config.yaml -v ./config_defaults.yaml -k "(\!rocoto\b)"
+        # Generate workflow files ...
+        ./generate_FV3LAM_wflow.py
+cd ${workspace}
+
+cd ${EXPTDIR}
+pwd
+cp ${workspace}/ush/wrappers/* .
+
+# Set parameters that the task scripts require ...
+export JOBSdir=${workspace}/jobs
+export USHdir=${workspace}/ush
+export PDY=20190615
```

Comment:
is PDY mean to stay fixed?

Reply:
PDY here is what is used for the default sample. If we want to allow a different sample, then we need a few of these do be variables.
Note: some of these definitions, JOBSdir, USHdir, ... were missing from the docs, and were discovered by trial+error.